src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/UserProfileFunctionalTest.java

```diff
@@ -0,0 +1,356 @@
+
+        assertThat(resourceForDeleteCheck.getRoles().size()).isNotNull();
+        assertThat(resourceForDeleteCheck.getRoles().size()).isEqualTo(3);
+        assertThat(resourceForDeleteCheck.getRoles().contains("pui-finance-manager,pui-case-manager,pui-user-manager"));
+        assertThat(!resourceForDeleteCheck.getRoles().contains(puiOrgManager));
+
+        log.info("deleteRolesOfActiveUserProfileShouldReturnSuccess :: ENDED");
+    }
+
+    public void reinviteUserWhenReinvitedWithinOneHourShouldReturn429() throws JsonProcessingException {
+        log.info("reinviteUserReturn429WhenReinvitedWithinOneHour :: STARTED");
+
+        UserProfileCreationData data = createUserProfileDataWithReInvite();
+        data.setEmail(pendingUserProfileCreationData.getEmail());
+
+        ErrorResponse errorResponse = testRequestHandler.sendPost(
+                testRequestHandler.asJsonString(data),
+                HttpStatus.TOO_MANY_REQUESTS,
+                requestUri).as(ErrorResponse.class);
+
+        assertThat(errorResponse.getErrorMessage()).isEqualTo(
+                String.format("10 : The request was last made less than %s minutes ago. Please try after some time",
+                        resendInterval));
+
+        log.info("reinviteUserReturn429WhenReinvitedWithinOneHour :: ENDED");
+    }
+
+    public void deleteActiveAndPendingUserShouldReturnSuccess() throws JsonProcessingException {
+        log.info("deleteActiveUserShouldReturnSuccess :: STARTED");
+
+        List<String> userIds = asList(activeUserProfile.getIdamId(), pendingUserProfile.getIdamId());
+
+        UserProfileDataRequest deletionRequest = buildUserProfileDataRequest(userIds);
+
+        testRequestHandler.sendDelete(
+                objectMapper.writeValueAsString(deletionRequest),
+                NO_CONTENT, requestUri);
+
+        testRequestHandler.sendGet(NOT_FOUND,
+                requestUri + "?userId=" + userIds.get(0));
+
+        log.info("deleteActiveUserShouldReturnSuccess :: ENDED");
+    }
+
+    public void unauthenticatedRequestsShouldReturn401() {
+        endpoints.forEach(callbackEndpoint ->
+                SerenityRest.given()
+                        .relaxedHTTPSValidation()
+                        .baseUri(targetInstance)
+                        .when()
+                        .get(callbackEndpoint)
+                        .then()
+                        .statusCode(HttpStatus.UNAUTHORIZED.value()));
+    }
+
+    public void invalidServiceAuthorisationRequestsShouldReturn401() {
+        endpoints.forEach(endpoint ->
+                SerenityRest.given()
+                        .relaxedHTTPSValidation()
+                        .baseUri(targetInstance)
+                        .header("ServiceAuthorization", "invalidServiceToken")
+                        .when()
+                        .get(endpoint)
+                        .then()
+                        .statusCode(HttpStatus.UNAUTHORIZED.value()));
+    }
```

Comment:
could you please add 1 forbidden/403 test case scenario ?

Reply:
Addressed here: https://github.com/hmcts/rd-user-profile-api/pull/621/commits/cde4987a95d5136a30cffabd2da8e8634a0b412a